nixos/modules/programs/x2goserver.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.x2goserver;
+in {
+  options.programs.x2goserver.enable = mkEnableOption "x2goserver";
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.x2goserver ];
+    users.extraGroups.x2gouser = {};
+    users.extraUsers.x2gouser = {
+      createHome = true;
```

Comment:
See comment below about `systemd` service. This could be replaced by `StateDirectory = "x2go";` if the `systemd` service was used.

Reply:
The activation script is still necessary even with a cleanup systemd-service, see below.

Reply:
Even still you should `StateDirectory` instead. `createHome` is also kind of bad because you can't control the permissions of the home directory.

Reply:
StateDirectory now and createHome gone.